### ISSUE TITLE
docker exec is throwing "rpc error: code = 13 desc = transport is closing", when daemon started with --live-restore

### ISSUE THREAD

**Issue by manoj0077** [docker/docker#28470]
**Description**

When daemon is started with --live-restore, seeing issues with docker exec after dockerd is restarted.

**Steps to reproduce the issue:**
1. Start daemon with --live-restore
2. Create a container and restart the docker daemon
3. docker exec of previous container 

**Describe the results you received:**

panic: assignment to entry in nil map

goroutine 51 [running]:
panic(0x8b4720, 0xc420252e80)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/docker/containerd/supervisor.(*Supervisor).newExecSyncChannel(0xc42015ea90, 0xc4202eb440, 0x40, 0xc4202eb4c0, 0x40)
        /tmp/tmp.4MHbXm2v1h/src/github.com/docker/containerd/supervisor/supervisor.go:418 +0xe7
github.com/docker/containerd/supervisor.(*Supervisor).addProcess(0xc42015ea90, 0xc42013a480, 0x0, 0xc4201dff60)
        /tmp/tmp.4MHbXm2v1h/src/github.com/docker/containerd/supervisor/add_process.go:39 +0x280
github.com/docker/containerd/supervisor.(*Supervisor).handleTask(0xc42015ea90, 0xd6b880, 0xc42013a480)
        /tmp/tmp.4MHbXm2v1h/src/github.com/docker/containerd/supervisor/supervisor.go:378 +0x253
github.com/docker/containerd/supervisor.(*Supervisor).Start.func1(0xc42015ea90)
        /tmp/tmp.4MHbXm2v1h/src/github.com/docker/containerd/supervisor/supervisor.go:282 +0xa0
created by github.com/docker/containerd/supervisor.(*Supervisor).Start
        /tmp/tmp.4MHbXm2v1h/src/github.com/docker/containerd/supervisor/supervisor.go:284 +0x3a4
time="2016-11-16T08:04:32.294695120Z" level=error msg="Error running exec in container: rpc error: code = 13 desc = transport is closing"
time="2016-11-16T08:04:32.294891144Z" level=error msg="libcontainerd: failed to receive event from containerd: rpc error: code = 13 desc = transport is closing"
time="2016-11-16T08:04:32.295087294Z" level=error msg="Handler for POST /v1.25/exec/517cf24f303a0f344d69bbbf9c7e2b5382f18bb99eab1241e3d66e3a3d43b443/resize returned error: rpc error: code = 14 desc = grpc: the connection is unavailable"
time="2016-11-16T08:04:32.591035298Z" level=info msg="libcontainerd: new containerd process, pid: 13859"

**Output of `docker version`:**

root@eebfd79a59ef:/go/src/github.com/docker/docker# docker version
Client:
 Version:      1.13.0-rc1
 API version:  1.25
 Go version:   go1.7.3
 Git commit:   2340df3-unsupported
 Built:        Wed Nov 16 07:15:36 2016
 OS/Arch:      linux/amd64

Server:
 Version:             1.13.0-rc1
 API version:         1.25
 Minimum API version: 1.12
 Go version:          go1.7.3
 Git commit:          2340df3-unsupported
 Built:               Wed Nov 16 07:15:36 2016
 OS/Arch:             linux/amd64
 Experimental:        false

**Output of `docker info`:**

root@eebfd79a59ef:/go/src/github.com/docker/docker# docker info
Containers: 1
 Running: 1
 Paused: 0
 Stopped: 0
Images: 1
Server Version: 1.13.0-rc1
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 3
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge epbind host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 8517738ba4b82aff5662c97ca4627e7e4d03b531
runc version: ac031b5bf1cc92239461125f4c1ffb760522bbf2
init version: N/A (expected: v0.13.0)
Security Options:
 apparmor
Kernel Version: 3.16.0-30-generic
Operating System: Ubuntu 14.04.5 LTS (containerized)
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 7.798 GiB
Name: eebfd79a59ef
ID: P5LM:WHR5:E5HK:VR2S:TKFF:KQAB:OEOE:BAVL:XYPQ:FMYL:TFXY:U5CW
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: true

**Comment by coolljt0725** [comment]
@manoj0077  Thanks for your report

I can reproduce this, ping @mlaventure

**Comment by thaJeztah** [comment]
@mlaventure reopening this because I don't think that containerd version was vendored yet

**Comment by mlaventure** [comment]
@thaJeztah thanks! It wasn't vendored yet indeed. Doing the PR today.

**Comment by ichekrygin** [comment]
We are hitting this issue in `1.12.6` with k8s `v1.6.2`
```
Apr 29 18:47:22 ip-10-72-133-177 dockerd[3163]: time="2017-04-29T18:47:22.110373177Z" level=error msg="libcontainerd: failed to receive event from containerd: rpc error: code = 13 desc = transport is closing
```

**Comment by thaJeztah** [comment]
@ichekrygin the fix for this issue is in docker 1.13 and above (including 17.03, 17.04)
